frontend/public/components/monitoring/alerting.tsx

```diff
@@ -599,10 +624,81 @@ export const AlertsDetailsPage = withFallback(
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const labels: PrometheusLabels = React.useMemo(() => alert?.labels, [labelsMemoKey]);
 
+    const AlertStateDescription = ({ alertData }) => {
+      if (alertData && !_.isEmpty(alertData.silencedBy)) {
+        return (
+          <StateTimestamp
+            text={t('public~Ends')}
+            timestamp={_.max(_.map(alertData.silencedBy, 'endsAt'))}
+          />
+        );
+      }
+      if (alertState && alertState.activeAt) {
+        return <StateTimestamp text={t('public~Since')} timestamp={alertData.activeAt} />;
+      }
+      return null;
+    };
+
+    const getSourceKey = (source) => {
```

Comment:
As with `getStateKey`, I think it would be preferable to move `getSourceKey` outside of the components so it can be shared.